src/main/java/com/engine/algorithm/SymbolicRegressionGE.java

```diff
@@ -75,6 +79,9 @@ public class SymbolicRegressionGE extends AbstractProblemGE {
     private String objective;
     private boolean de = false;
 
+    @Autowired
```

Comment:
@Autowired no suele funcionar en clases no gestionadas por Spring, ¿has comprobado que la dependencia se inyecte correctamente?

Reply:
Como estamos utilizando Spring Boot, por defecto el automatic component scanning ya esta activado, por esa parte no tienes que preocuparte.
Seria suficiente con, primero, como dices, marcarla con `@Component`, `@Service`, o cualquier otra anotación similar, y segundo, **nunca crearla con new**, donde sea necesaria pedirla con `@Autowired` o por el constructor.

Reply:
He intentado utilizar las anotaciones.

Sin embargo daba error pues me pedia definir un bean de tipo int (refiriendose al segundo parametro del constructor de SymbolicRegressionGE).

De momento la solucion que he encotrado es esta: https://confluence.jaytaala.com/display/TKB/Super+simple+approach+to+accessing+Spring+beans+from+non-Spring+managed+classes+and+POJOs

Basicamente crea una clase que implementa ApplicationContextAware. 
En ella define un metodo static que sirve para obtener cualquier bean.

En lugar de utilizar @Autowired para el rabbitTemplate, llamo a tal metodo.

¿Como lo ves?

Reply:
@rmartinsanta

Reply:
Yo optaría por una solución más fácil, eliminar el Autowired y desde donde llames a SymbolicRegressionGE, pasarle el RabbitTemplate a través del constructor. Es posible que desde donde lo llames sí sea una clase gestionada por Spring, y por lo tanto, puedes pedirle a Spring que te lo inyecte. ¿Me explico?